apps/ui/lib/ui/channels/game_channel.ex

```diff
@@ -14,13 +14,129 @@ defmodule Ui.GameChannel do
 
   def handle_info({:after_join, shortcode}, socket) do
     summary = GameApp.Server.summary(shortcode)
-
     push(socket, "game_summary", summary)
-
     {:noreply, socket}
   end
 
+  def handle_in("start_game", _params, socket) do
+    {shortcode, player} = game_context(socket)
+
+    handle_creator_action(shortcode, player, socket, fn ->
+      GameApp.Server.start_game(shortcode)
+      broadcast_game_state(shortcode, socket)
+    end)
+  end
+
+  def handle_in("start_round", _params, socket) do
+    {shortcode, player} = game_context(socket)
+
+    handle_funmaster_action(shortcode, player, socket, fn ->
+      GameApp.Server.start_round(shortcode)
+      broadcast_game_state(shortcode, socket)
+    end)
+  end
+
+  def handle_in("select_prompt", %{"prompt" => prompt}, socket) do
+    {shortcode, player} = game_context(socket)
+
+    handle_funmaster_action(shortcode, player, socket, fn ->
+      GameApp.Server.select_prompt(shortcode, prompt)
+      broadcast_game_state(shortcode, socket)
+    end)
+  end
+
+  def handle_in("select_reaction", %{"reaction" => reaction}, socket) do
+    {shortcode, player} = game_context(socket)
+
+    handle_action(shortcode, socket, fn ->
+      GameApp.Server.select_reaction(shortcode, player, reaction)
+      broadcast_game_state(shortcode, socket)
+    end)
+  end
+
+  def handle_in("select_winner", %{"winner" => winner}, socket) do
+    {shortcode, player} = game_context(socket)
+
+    handle_funmaster_action(shortcode, player, socket, fn ->
+      GameApp.Server.select_winner(shortcode, winner)
+      broadcast_game_state(shortcode, socket)
+    end)
+  end
+
+  # private
+
   defp current_player(socket) do
     socket.assigns.current_player
   end
+
+  defp handle_creator_action(shortcode, player, socket, callback) do
+    with {:ok, _pid} <- find_server(shortcode),
+         {:ok, summary} <- game_summary(shortcode),
+         {:ok} <- user_is_creator(player, summary) do
+      callback.()
```

Comment:
This syntax is wonky 😢 

(Not saying its wrong or theres a better way, just complaining)

Reply:
Yea for sure. I'd ideally just pass a block if I could yield but I don't think that's a thing